### PR TITLE
feat(SAI): generate single comment

### LINKED ISSUE #2467: SAI Comments not triggering a diff
If you update a SAI script with only Generate comments it will not provide a new diff and the new comment will not be updated. You need to change an actual param for the comment to update in the diff

### PR DESCRIPTION
**Pull request by Helias**
- closes https://github.com/azerothcore/Keira3/issues/2107

I've added the possibility to generate a single comment OR all the comments of all rows at once

image

### CODE REVIEW

File: libs/shared/sai-editor/src/sai-editor.service.spec.ts [hunk 1]
```diff
@@ -144,5 +144,31 @@ describe('SAI Editor Service', () => {
...
+
+      service.generateComments();
+      tick(100);
+
+      expect(updateDiffQuerySpy).toHaveBeenCalledTimes(1);
+      expect(updateFullQuerySpy).toHaveBeenCalledTimes(1);
+      expect(refreshDatatableSpy).toHaveBeenCalledTimes(1);
+      expect(isRowSelectedSpy).toHaveBeenCalledTimes(1);
+      expect(getNameSpy).toHaveBeenCalledTimes(1);
+      expect(generateCommentSpy).toHaveBeenCalledTimes(1);
```
Comment (FrancescoBorzi) [inline libs/shared/sai-editor/src/sai-editor.service.spec.ts:171]: shouldn't we also check on the actual comment that has been added?

File: libs/shared/sai-editor/src/sai-editor.component.html [hunk 2]
```diff
@@ -557,8 +557,15 @@
...
-            Generate comments
+          <button id="generate-comments-btn" class="btn btn-primary btn-sm float-end" (click)="editorService.generateComments(true)">
+            Generate comments (all)
+          </button>
+          <button
+            id="generate-comments-btn-single"
+            class="btn btn-primary btn-sm float-end me-2"
+            (click)="editorService.generateComments()"
+          >
+            Generate comment
```
Comment (FrancescoBorzi) [inline libs/shared/sai-editor/src/sai-editor.component.html:568]: shouldn't we use translations? also the text should be more specific, like "Generate comment for selected row"

Comment (FrancescoBorzi) [reply libs/shared/sai-editor/src/sai-editor.component.html:568]: also this button should be disabled if no row is selected

File: libs/shared/sai-editor/src/sai-editor.service.ts [hunk 3]
```diff
@@ -93,22 +93,35 @@ export class SaiEditorService extends MultiRowComplexKeyEditorService<SmartScrip
...
   }
+
+  private async generateSingleComment(row: SmartScripts): Promise<string> {
+    return this.saiCommentGeneratorService.generateComment(
+      structuredClone(this._newRows),
+      { ...row },
+      /* istanbul ignore next */
+      await this.handlerService.getName()?.toPromise(),
+    );
+  }
```
Comment (FrancescoBorzi) [inline libs/shared/sai-editor/src/sai-editor.service.ts:126]: missing integration tests

File: libs/shared/sai-editor/src/sai-editor.service.spec.ts (hunk 1 above)
Comment (Helias) [reply libs/shared/sai-editor/src/sai-editor.service.spec.ts:171]: everything is mocked, so the comment is undefined here

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**